src/backend/distributed/planner/deparse_shard_query.c

```diff
@@ -90,11 +90,23 @@ RebuildQueryStrings(Query *originalQuery, List *taskList)
 			}
 		}
 
-		deparse_shard_query(query, relationId, task->anchorShardId,
-							newQueryString);
+		/*
+		 * UPDATE and DELETE queries can include subqueries, so treat them
+		 * differently.
```

Comment:
Maybe clarify what's different between `deparse_shard_query` and `pg_get_query_def`. Might also be something we should address.